RFP-01.md

```diff
@@ -0,0 +1,175 @@
+# RFP-01: Settlement contract wrapper
+
+## Background
+
+The CoW DAO has recently passed [CIP-44: Reduced bonding requirements](https://snapshot.org/#/cow.eth/proposal/0x1b6f1171633ec3d20c4370db37074aa1bd830486d4d0d6c26165915cc42d9412), which allows solvers to participate in the batch auction with reduced bonding requirements. The reduced bonding requirements are as follows:
+
+- $50k in yield-bearing stable coins or ETH
+- 500k COW
+
+The target deposit is:
+
+- $100k in yield-bearing stable coins or ETH
+- 1M COW
+
+## Goal
+
+The goal of this project is to implement a `SubPoolFactory` that allows solvers to participate in the batch auction with reduced bonding requirements. The `SubPoolFactory` will deploy a `SubPool` for each solver, which will be funded to at least the minimum required amount. The `SubPoolFactory` will also provide a wrapper around the `GPv2Settlement` contract that allows solvers to execute their solutions by providing a signature attesting to the solution and their parameters.
+
+### Key Features
+
+- Ultra gas efficient signed settlement function
+- No co-mingling of funds between solvers
+- Easily observable on-chain addresses for MEV Blocker rebates
+- Permissionless access to the batch auction with reduced bonding requirements
+
+### Deliverables
+
+- Contract system for `SubPoolFactory`, `SubPool`, and `SignedSettlement`
+- Comprehensive documentation
+- Open-source code repository
+
+### Interface
+
+```solidity
+interface Auth {
+    /**
+     * Allows `usr` to perform root-level functions
+     */
+    function addOwner(address usr) external auth;
+    /**
+     * Denies `usr` from being able to perform root-level functions
+     */
+    function removeOwner(address usr) external auth;
+    /**
+     * Determine if `usr` is able to perform root-level functions
+     */
+    function isOwner(address usr) external view returns (bool);
+}
+
+interface SubPool is Auth {
```

Comment:
Why does a subpool need to exist? Can't everything be done with internal accounting in the factory?

Reply:
This is done so that MEV blocker kickbacks can be sent to the address which simplifies accounting instead of this being all co-mingled. This is also somewhat of an oversight (ETH handling) and something that I'd prefer to kick back to the grantee to work out how to handle the additional collateral.

Reply:
In principle this can be handled as well with internal balances (`deposit(target[], amount[])`. But if it's easier to integrate this with other tools, I'm ok with keeping subpool. We should still add the ETH handling (and arbitrary ERC20 as well?) to the scope.

Reply:
> In principle this can be handled as well with internal balances (deposit(target[], amount[])). But if it's easier to integrate this with other tools, I'm ok with keeping subpool. We should still add the ETH handling (and arbitrary ERC20 as well?) to the scope.

In principle, yes, but this requires more integration, no? If having a subpool address, the solver can simply specify the sub pool address in the MEV blocker RPC URL and this will cause the rebate to be sent there (just a simple transfer). Then no co-mingling of funds, or requirement for craz(ier) accounting?